File: lua/difference-of-squares/difference-of-squares.lua
local function square_of_sum(n)
	local count = 0
	for i = 1, n do
		count = count + i
	end
	return count * count
end

local function sum_of_squares(n)
	local count = 0
	for i = 1, n do
		count = count + i * i
	end
	return count
end

local function difference_of_squares(n)
	return square_of_sum(n) - sum_of_squares(n)
end

return {
	square_of_sum = square_of_sum,
	sum_of_squares = sum_of_squares,
	difference_of_squares = difference_of_squares,
}

File: typescript/difference-of-squares/difference-of-squares.ts
export class Squares {
  private nums: number[]
  constructor(count: number) {
    this.nums = Array.from({ length: count }, (_, i) => i + 1)
  }

  get sumOfSquares(): number {
    return this.nums.reduce((acc, n) => acc + n * n, 0)
  }

  get squareOfSum(): number {
    const sum = this.nums.reduce((acc, n) => acc + n, 0)
    return sum * sum
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}
